### ISSUE TITLE
Add keyboard shortcuts

### ISSUE THREAD

**Issue by PopeRigby** [LemmyNet/lemmy#281]
Some ideas:
* Create link = (Highlight text) + Ctrl + K
* New post = Ctrl + N

**Comment by zacanger** [comment]
Whenever a site has built-in keyboard shortcuts, that adds the risk of interfering with default browser behavior (which can break accessibility), and also with browser extensions (for us Vimmers out there). Not saying it's a bad idea, just something to keep in mind and be careful of.

**Comment by dessalines** [comment]
Same, I use vim-like browser plugins, and if I'm lucky, any sites that use hotkeys will let me disable them.

**Comment by PopeRigby** [comment]
Of course. There should be an option to turn them off.

**Comment by ptman** [comment]
And a way to show the current shortcuts (link + shortcut)

**Comment by Atemu** [comment]
I think both vim- and emacs-like keybinds would be great to have and they could probably even exist side-by-side.  

It'd be really cool if you could navigate the whole UI with keyboard only, that's something you can't even do with RES. It would also make the site more accessible for people who cannot easily use a mouse.

Ideally the specific shortcuts would be fully configurable.

**Comment by dessalines** [comment]
> It'd be really cool if you could navigate the whole UI with keyboard only

What's even cooler, is navigating *any* site with a keyboard, which many of us currently do, using plugins like vim-vixen, trydactyl, etc.

This is why I'm probably not going to add keyboard shortcuts to lemmy. This is a solved problem for the web in general, and any keyboard shortcuts would just conflict with that solution.

**Comment by Atemu** [comment]
I use Vimium myself too but I find site-specific shortcuts like RES' much more convenient and useful. 

Browsing typically works fine with external tools but interacting with the website (collapse replies, open post, up/downvote, reply, save etc.) is a lot better with "native" shortcuts rather than "pressing" the website's buttons with a long key combination.
